feature-libs/product-configurator/rulebased/root/interceptor/cpq-configurator-rest.interceptor.ts

```diff
@@ -0,0 +1,104 @@
+import {
+  HttpErrorResponse,
+  HttpEvent,
+  HttpHandler,
+  HttpInterceptor,
+  HttpRequest,
+  HttpResponse,
+} from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Observable, throwError } from 'rxjs';
+import { catchError, switchMap, take, tap } from 'rxjs/operators';
+
+import { CpqAccessData } from './cpq-access-data.models';
+import { CpqAccessStorageService } from './cpq-access-storage.service';
+
+const HEADER_ATTR_CPQ_SESSION_ID = 'x-cpq-session-id';
+const HEADER_ATTR_CPQ_NO_COOKIES = 'x-cpq-disable-cookies';
+
+export const CPQ_CONFIGURATOR_VIRTUAL_ENDPOINT =
```

Comment:
The concept of the virtual endpoint is interesting and very custom at the same time. I hope we can solve it with more standard approach. Idea from top of my head (I need to think about it yet): create the `CpqEndpointsService` similar to `OccEndpointsService`, which prepends the CPQ base url, the configurable API prefix `backend.cpq.prefix` and the configurable endpoints taken from config `backend.cpq.endpoints`). Does it make sense for you?

Reply:
@Uli-Tiger Could you take a look into that?

Reply:
Now, all things considered, I think it would be more clean to set an artificial placeholder http header instead of the placeholer endpoint prefix. Then in the interceptor could recognize requests by this specific header and remove it before passing the request along.


The different thing is that specific endpoints are currently hardcoded in the `CpqConfiguratorRestService`. Moreover the part `/api/configuration/v1/` is repeated in all of them. Although it's not a big problem, for consistency it would be just nice, to follow the same pattern as OCC endpoints:
- backend endpoints configurable via `backend.cpq.endpoints`
- backend prefix configurable via `backend.cpq.prefix`
- endpoints accessed by name via a new service `CpqEndpointsService` 

I'm aware `CpqEndpointsService` cannot set the base url of the CPQ server, because it's not known early, but only in the interceptor, taken from `CpqAccessData`

Reply:
pick